src/software/simulation/physics/physics_ball.cpp

```diff
@@ -189,6 +194,87 @@ void PhysicsBall::updateIsInFlight() const
     }
 }
 
+void PhysicsBall::setInitialKickSpeed(double speed)
+{
+    initial_kick_speed = speed;
+}
+
+void PhysicsBall::applyBallFrictionModel(const Duration &time_step)
+{
+    // Friction model adapted from section 5 of
+    // https://ssl.robocup.org/wp-content/uploads/2020/03/2020_ETDP_ZJUNlict.pdf
```

Comment:
I think its worth summarizing how you achieved this from reading that paper

Reply:
added a comment